hbase-client/src/main/java/org/apache/hadoop/hbase/client/MetricsConnection.java

```diff
@@ -637,14 +637,17 @@ private void shutdown() {
   }
 
   /** Report RPC context to metrics system. */
-  public void updateRpc(MethodDescriptor method, Message param, CallStats stats) {
+  public void updateRpc(MethodDescriptor method, Message param, CallStats stats, boolean failed) {
     int callsPerServer = stats.getConcurrentCallsPerServer();
     if (callsPerServer > 0) {
       concurrentCallsPerServerHist.update(callsPerServer);
     }
     // Update the counter that tracks RPCs by type.
     final String methodName = method.getService().getName() + "_" + method.getName();
     getMetric(CNT_BASE + methodName, rpcCounters, counterFactory).inc();
+    if (failed) {
+      getMetric(CNT_BASE + methodName + "_Failure", rpcCounters, counterFactory).inc();
```

Comment:
Thoughts on just creating a new `FAILURE_CNT_BASE` with value `rpcFailureCount_`? This would be one less concatenation and also more intuitive name and easier to parse for consumers.

Reply:
Updated with your suggestion, please take a look. Thanks!